src/client.ts

```diff
@@ -338,21 +370,22 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     options: optimizely.OptimizelyDecideOption[] = [],
     overrideUserId?: string,
     overrideAttributes?: optimizely.UserAttributes
-  ): OptimizelyDecision {    
+  ): OptimizelyDecision {
     const user = this.getUserContextWithOverrides(overrideUserId, overrideAttributes);
     if (user.id === null) {
       logger.info('Not Evaluating feature "%s" because userId is not set', key);
       return createFailedDecision(key, `Not Evaluating flag ${key} because userId is not set`, user);
-    }    
-    const optlyUserContext: optimizely.OptimizelyUserContext | null = this._client.createUserContext(user.id, user.attributes);
+    }
+
+    const optlyUserContext = this.getUserContextInstance(user);
```

Comment:
When the user provides override user and attributes to the decide call, this logic will replace the memoized user context. In my opinion.
1. Memoized userContext should only be replaced/recreated when an explicit call to `setUser` is made i.e the `user` props to `OptimizelyProvider` changes.
2. If override User Id and / or attributes are used in a call to hook or eventaully to `decide` method, it should create a temporary userContext and use that in context of that particular call.

Overriding is temporary, it should not replace the permanently memoized user passed in as prop to OptimizelyProvider

Reply:
Ignore this comment, i just realized you are not overriding the memoized context in the method.